collection-scripts/gather_monitoring

```diff
@@ -1,29 +1,235 @@
+        return
+    fi
+
+    echo "INFO: Metrics will be collected from [$(date -d "@${DATE_START}")] to [$(date -d "@${DATE_END}")]"
+    for METRIC in $GATHER_MONIT_CUSTOM_METRICS; do
+        echo "INFO: Getting metric range: $METRIC"
+        prom_query_range "$METRIC" "$METRIC"
+    done
+}
+
+# Discovery metrics from prefixes defined on environment GATHER_MONIT_DISCOVERY_PREFIXES
+function get_discovery_metrics() {
+
+    if [[ -z "${GATHER_MONIT_DISCOVERY_PREFIXES}" ]]; then
+        return
+    fi
+
+    echo "INFO: Starting discovery metrics by prefixes..."
+    # TODO: fix it shame~~ O(n*m) ;(
+    while read METRIC; do
+        for PREFIX in ${GATHER_MONIT_DISCOVERY_PREFIXES}; do
+            if [[ "${METRIC}" =~ ^${PREFIX}.* ]]; then
+                echo "INFO: Getting metric range: ${METRIC}"
+                prom_query_range "${METRIC}" "${METRIC}" ;
+            fi
+        done
+    done < <( curl -sk ${CURL_OPTS} \
+        -H "Authorization: Bearer ${OCP_TOKEN}" \
+        -H 'Accept-encoding: gzip' \
+        -g "${PROM_URL}/api/v1/label/__name__/values" \
+        |gunzip \
+        | jq -r '.data[]'; \
+    )
+    echo "INFO: Discovery metrics by prefixes finished"
+}
+
+# List all Dashboards from default folder (id=1) on Grafana - where OCP store it's dashboards
+function gf_discovery_dashboards() {
+    echo "INFO: Discovery Dashboards from default Folder"
+    curl ${CURL_OPTIONS} \
+        -H "Authorization: Bearer ${OCP_TOKEN}" \
+        "${GF_URL}/api/search?folderIds=1" > ${GF_PATH}/dashboards.json
+}
+
+# Parse and download all Dashboards discovered from Default folder,
+# then collect it's metrics from Prometheus API.
+function gf_discovery_metrics_from_dashboards() {
+
+    install_jq
+    ${JQ_PATH} \
+        -r '.[] | .uri +":"+ .uid' ${GF_PATH}/dashboards.json \
+        | awk -F'db/' '{print$2}' \
+        | awk '{print$1":"$2}' > ${GF_PATH}/dashboards.txt
+
+    while read DASHBOARD; do
+        NAME=$(echo $DASHBOARD |awk -F':' '{print$1}')
+        DUID=$(echo $DASHBOARD |awk -F':' '{print$2}')
+
+        echo "INFO: Getting Dashboard ${NAME}"
+        curl ${CURL_OPTIONS} \
+            -H "Authorization: Bearer ${OCP_TOKEN}" \
+            $GF_URL/api/dashboards/uid/${DUID} > ${GF_PATH}/dashboard_${NAME}.json
+
+        ${JQ_PATH} \
+            .dashboard.rows[].panels[].targets[].expr ${GF_PATH}/dashboard_${NAME}.json \
+            | egrep -o '\((\w+){\)*' \
+            | tr -d '({' > ${GF_PATH}/dashboard_${NAME}_metrics.txt
+    done < ${GF_PATH}/dashboards.txt
+
+    echo "INFO: Extracting metrics from dashboards"
+    sort -u ${GF_PATH}/dashboard_*_metrics.txt > ${GF_PATH}/dashboards_metrics.txt
+    while read METRIC; do
+        echo "INFO: Getting metric range: $METRIC"
+        prom_query_range "$METRIC" "$METRIC"
+    done < ${GF_PATH}/dashboards_metrics.txt
+}
+
+function get_grafana_metrics() {
+    ENABLE_DISCOVERY=${GATHER_MONIT_DISCOVERY_GRAFANA:-"yes"}
```

Comment:
nit: in place of the default value of explicit "yes", you should perhaps use `GATHER_MONIT_DISCOVERY_GRAFANA_DEFAULT`.